opencast-ilias/OpenCast
# [Release 8] Upload Thumbnail (Preview)

This PR fixes #291,

### Description
When uploading a video, user is now able to either upload a thumbnail file or select a timepoint in the video to extract thumbnail from.

### How it works
#### Configurations
- In Plugins configuration, there is now a new sub tab under Settings > Thumbnails
- In that config page:
   - admin should enable the thumbnail upload
   - admin needs to select the accepted file types for the thumbnail file to be uploaded
   - admin needs to select a mode to provide the possible ways of uploading thumbnail (either by timepoint or file upload)

#### Opencast Configuration
This feature depends on installing new workflows and encoding profiles in the Opencast System. Please use this document for more info: 

[Documentation_ Upload of thumbnail.pdf](https://github.com/user-attachments/files/15915449/Documentation_.Upload.of.thumbnail.pdf)

###### Encoding
- /etc/opencast/encoding/manual-thumb.properties

###### fast workflow (if in use)
- /etc/opencast/workflows/fast-with-thumbnail-support.xml 

###### schedule-and-upload workflow (if in use)
- /etc/opencast/workflows/schedule-and-upload-with-thumbnail-support.xml
- /etc/opencast/workflows/partial-publish-with-thumbnail-support.xml

#### Upload page
- There is a new section in upload page called "Thumbnail"
- depending on the admin selection for mode, users will see:
   - a switchable group of ui components if "Both modes" is selected, by which users can only select one way: either by uploading a file via File Input or selecting a timepoint via Timepicker.
   - Only a file input if "File Upload" is selected
   - Only a Timepicker input if the "Timepoint" is selected.
- The "straightToPublishing" workflow param value from the configuration is also reflected in this page, by providing related infos or disabling the section if the value is set to always inactive!

**NOTE:** As for uploading the file, we had to use ingest mechanism, therefore, you would need to make sure that the uplaod by ingest is configured in your Opencast!

### How to test:
- Opencast 15 and above.
- ILIAS 8
- Patch this PR.
- Configuring the thumbnails settings in plugin config. (Check different options and their outcomes)
- Upload video:
  - Upload thumbnail
  - Select a timepoint (the timepoint is smart, which means it gets the duration of the selected video and does not allow you to go beyond the duration by each you select a video)!
  - Both modes should also cover the above scenarios.

**NOTE:** Please help adjust or improve the German texts!
